bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/cache/ExpiringCacheMapTest.java

```diff
@@ -13,188 +13,197 @@
+    private static final Supplier<String> CACHE_ACTION = () -> RandomStringUtils.random(8);
+    private static final Supplier<String> PREDICTABLE_CACHE_ACTION_1 = () -> RESPONSE_1;
+    private static final Supplier<String> PREDICTABLE_CACHE_ACTION_2 = () -> RESPONSE_2;
 
-    public static final Supplier<String> CACHE_ACTION = () -> RandomStringUtils.random(8);
-
-    public static final String FIRST_TEST_KEY = "FIRST_TEST_KEY";
-    public static final String SECOND_TEST_KEY = "SECOND_TEST_KEY";
+    private static final String FIRST_TEST_KEY = "FIRST_TEST_KEY";
+    private static final String SECOND_TEST_KEY = "SECOND_TEST_KEY";
 
+    private ExpiringCacheMap<String, String> subject;
+    
+    @Before
+    public void setUp() {
+        subject = new ExpiringCacheMap<>(CACHE_EXPIRY);
+    }
+    
+    
     @Test(expected = IllegalArgumentException.class)
-    public void testAddIllegalArgumentException1() throws IllegalArgumentException {
-        final ExpiringCacheMap<String, String> cache = new ExpiringCacheMap<>(CACHE_EXPIRY);
+    public void testPutIllegalArgumentException1() throws IllegalArgumentException {
+        subject.put(FIRST_TEST_KEY, (Supplier<String>) null);
+    }
 
-        final Supplier<String> action = null;
-        cache.put(FIRST_TEST_KEY, action);
+    @Test(expected = IllegalArgumentException.class)
+    public void testPutIllegalArgumentException2() throws IllegalArgumentException {
+        subject.put(FIRST_TEST_KEY, (ExpiringCache<String>) null);
     }
 
     @Test(expected = IllegalArgumentException.class)
-    public void testAddIllegalArgumentException2() throws IllegalArgumentException {
-        final ExpiringCacheMap<String, String> cache = new ExpiringCacheMap<>(CACHE_EXPIRY);
+    public void testPutIllegalArgumentException3() throws IllegalArgumentException {
+        subject.put(null, CACHE_ACTION);
+    }
+    
+    @Test
+    public void testPut() {
+        subject.put(FIRST_TEST_KEY, PREDICTABLE_CACHE_ACTION_1);
 
-        final ExpiringCache<String> item = null;
-        cache.put(FIRST_TEST_KEY, item);
+        assertEquals(RESPONSE_1, subject.get(FIRST_TEST_KEY));
+
+        subject.put(FIRST_TEST_KEY, PREDICTABLE_CACHE_ACTION_2);
+
+        assertEquals(RESPONSE_2, subject.get(FIRST_TEST_KEY));
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testAddIllegalArgumentException3() throws IllegalArgumentException {
-        final ExpiringCacheMap<String, String> cache = new ExpiringCacheMap<>(CACHE_EXPIRY);
+    @Test
+    public void testPutIfAbsentAndGet() {
+        String response = subject.putIfAbsentAndGet(FIRST_TEST_KEY, PREDICTABLE_CACHE_ACTION_1);
+
+        assertEquals(RESPONSE_1, response);
+        assertEquals(RESPONSE_1, subject.get(FIRST_TEST_KEY));
 
-        cache.put(null, CACHE_ACTION);
+        response = subject.putIfAbsentAndGet(FIRST_TEST_KEY, PREDICTABLE_CACHE_ACTION_2);
+
+        assertEquals(RESPONSE_1, response);
```

Comment:
I know that we as developer are responsible for the values of `RESPONSE_1` and `RESPONSE_2` because we are free to assign to them whatever we like. But this test case premises,  that those values are different. So let's add an `assertNotEquals(RESPONSE_1,RESPONSE_2)` - with an inline comment why we added it - to make sure the test will always work like expected. Wdyt?

Reply:
Yes, it crossed my mind for a while too that the test can succeed with less code ;-)

I will come up with something.